src/components/SyncView.js

```diff
@@ -51,20 +53,19 @@ export default class SyncView extends React.Component {
     if (!target) return
     this.destroyed = false
     this.stream = replicate.start(target)
-    this.stream.on('data', function (data) {
+    pump(this.stream, through.obj(function (data, enc, next) {
```

Comment:
Thinking on it, I don't think this will help solve the problem: `pump` also starts the pipeline, so data events may still be lost before someone starts using `this.stream` elsewhere.

Sorry, but I think your original solution was better, but combined with [stream#pause](https://nodejs.org/api/stream.html#stream_readable_pause) after the data event listener is set up. :)

Reply:
(Fun fact: if these were pull streams, it'd be easier to set up partial pipelines, since data never starts flowing until a sink is connected.)

Reply:
I wonder if this is an issue, because the stream won't be used by anyone else, only by this module